From b4ea9dd4b41142b4788b263e4656917032fcccf4 Mon Sep 17 00:00:00 2001
From: James Dinan <jdinan@nvidia.com>
Date: Wed, 6 Jul 2022 06:57:08 -0700
Subject: [PATCH 2/2] Temporary WAR for Slingshot provider linkspeed bug

The Slingshot provider underreports the link speed by a factor of 1,000. This
is a temporary workaround to fix it.

Signed-off-by: James Dinan <jdinan@nvidia.com>
---
 src/nccl_ofi_net.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/nccl_ofi_net.c b/src/nccl_ofi_net.c
index 5c46f02f..c18856e1 100644
--- a/src/nccl_ofi_net.c
+++ b/src/nccl_ofi_net.c
@@ -1308,6 +1308,8 @@ static ncclResult_t ofi_getProperties(int dev, ncclNetProperties_t *props)
 
 	/* Speed reported in Mbps */
 	dev_props.speed = nic_info->link_attr->speed / (1e6);
+	/* Temporary workaround to Slingshot provider link speed bug */
+	if (dev_props.speed < 1000) dev_props.speed *= 1000;
 
 	goto exit;
 
-- 
2.34.1

